### PR TITLE
check if image with never pullpolicy present on node in scheduler pre…

### PR DESCRIPTION
**Pull request by ChenLingPeng**
…dicate process

**What this PR does / why we need it**:
Inspired by #40639 , we should filter out those nodes which image not present but pull policy is `Never`.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: plugin/pkg/scheduler/algorithmprovider/defaults/defaults.go [hunk 1]
```diff
@@ -159,6 +159,9 @@ func defaultPredicates() sets.String {
 
 		// Fit is determined by node disk pressure condition.
 		factory.RegisterFitPredicate("CheckNodeDiskPressure", predicates.CheckNodeDiskPressurePredicate),
+
+		// Fir is determined by image present on node
```
Comment (k82cn) [inline plugin/pkg/scheduler/algorithmprovider/defaults/defaults.go]: s/Fir/Fit

**Comment by k82cn** [comment]
@k8s-bot ok to test

**Review by k82cn — COMMENTED** [review]
Make sense to me; let wait for @davidopp & @timothysc 's comments on `PullNever` policy.

**Comment by liggitt** [comment]
I don't think the list of images in node status is exhaustive (I think there is a max size to that list), so absence does not mean the image does not exist.

**Comment by liggitt** [comment]
I think this would incorrectly exclude nodes

**Comment by k82cn** [comment]
> I don't think the list of images in node status is exhaustive (I think there is a max size to that list), so absence does not mean the image does not exist.

Learned, is it restricted by `kubelet` or `docker`?

**Comment by liggitt** [comment]
By the kubelet. 

https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet_node_status.go#L584

Node status that included all images could be enormous and overwhelm the API and etcd

**Comment by liggitt** [comment]
Images in node status are advisory, not definitive

**Comment by k82cn** [comment]
> Node status that included all images could be enormous and overwhelm the API and etcd
> Images in node status are advisory, not definitive

Got that :). If so, I agree with this PR is incorrect and not necessary to enhance.

**Comment by ChenLingPeng** [comment]
Got it. Close this since it is not correct.

**Pull request closed without merging**